File: listen.cs
using System.Text;
using MQTTnet;
using MQTTnet.Client;

namespace MqttClientApp;

internal class Program
{
    private static async Task Main(string[] args)
    {
        // Create a new MQTT client.
        var factory = new MqttFactory();
        var mqttClient = factory.CreateMqttClient();

        // Configure options
        var options = new MqttClientOptionsBuilder()
            .WithClientId("ClientId")
            .WithTcpServer("SD.local", 1883) // Replace with your MQTT broker IP and port
            .Build();

        // Connect to the MQTT broker
        await mqttClient.ConnectAsync(options);

        Console.WriteLine("Connected successfully with MQTT Brokers.");
        // Subscribe to the topic
        var mqttFactory = new MqttFactory();
        var mqttSubscribeOptions = mqttFactory.CreateSubscribeOptionsBuilder()
                .WithTopicFilter(
                    f => { f.WithTopic("canbus/#"); }).Build()
            ;
        await mqttClient.SubscribeAsync(mqttSubscribeOptions);
        Console.WriteLine("Subscribed");

        mqttClient.ApplicationMessageReceivedAsync += e =>
        {
            Console.WriteLine($"Id = {e.ApplicationMessage.Topic.Split('/')[1]}");
            Console.WriteLine($"Payload = {Encoding.UTF8.GetString(e.ApplicationMessage.Payload)}");
            Console.WriteLine();
            return Task.CompletedTask;
        };

        Console.WriteLine("Press any key to exit.");
        Console.ReadLine();

        // Disconnect from the MQTT broker
        await mqttClient.DisconnectAsync();
    }
}

File: listen.py
import paho.mqtt.client as mqtt

def on_connect(client, userdata, flags, rc):
    print("Connected with result code " + str(rc))
    # client.subscribe("A1")
    # client.subscribe("A2")

    client.subscribe("canbus/#")

def on_message(client, userdata, msg):
    print(msg.topic + " " + str(msg.payload))

client = mqtt.Client()
client.on_connect = on_connect
client.on_message = on_message

client.connect("SD.local", 1883, 60)

client.loop_forever()
